plugins/modules/aci_l4l7_policy_based_redirect_destination.py

```diff
@@ -0,0 +1,404 @@
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        policy=dict(type="str", aliases=["policy_name"]),
+        ip=dict(type="str", aliases=["redirect_ip"]),
+        additional_ip=dict(type="str"),
+        mac=dict(type="str", aliases=["redirect_mac"]),
+        logical_device=dict(type="str", aliases=["logical_dev"]),
+        concrete_device=dict(type="str", aliases=["concrete_dev"]),
+        concrete_interface=dict(type="str", aliases=["concrete_intf"]),
+        destination_name=dict(type="str", aliases=["dest_name"]),
+        destination_type=dict(type="str", aliases=["dest_type"], choices=["l1/l2", "l3"]),
+        health_group=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        pod_id=dict(type="int"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[["state", "absent", ["tenant", "policy"]], ["state", "present", ["tenant", "policy"]]],
```

Comment:
can you format this on multiple lines? run black after the below change
```suggestion
        required_if=[["state", "absent", ["tenant", "policy"]], ["state", "present", ["tenant", "policy"]],],
```